xrpl/asyncio/clients/client.py

```diff
@@ -51,3 +54,24 @@ async def _request_impl(
         :meta private:
         """
         pass
+
+
+async def _get_network_id_and_build_version(client: Client) -> None:
```

Comment:
Why not automatically run this on every client upon instantiation, instead of having it be a helper function? Then it only needs to be run once, instead of every time a function that uses this is called.

Reply:
We are not duplicating invocations of this function. Here are [two](https://github.com/XRPLF/xrpl-py/blob/4779d796e6f808c952214e794be5d9d342412e14/xrpl/asyncio/transaction/main.py#L233) [examples](https://github.com/XRPLF/xrpl-py/blob/ea6365af1b3da29c45f4441fefcfb298f92d897d/xrpl/asyncio/wallet/wallet_generation.py#L62) where the `network_id` information persists in the client object.

But I agree, this could be done in a cleaner way.
I can only invoke `_get_network_id_and_build_version` function after a client has opened the socket-connection. So, I'll need to invoke this in this `open()` method of all the child classes of `Client`

let me try it out

Reply:
I'm open to the option of doing it lazily - i.e. only if and as needed. All that would require is checking if the network ID has been instantiated before running the request.

Reply:
okay, I've added a check: [23d3eb3](https://github.com/XRPLF/xrpl-py/pull/715/commits/23d3eb31df830a1c55f2cae51e720fde14c673a3)